internal/service/streamprocessor/model.go

```diff
@@ -51,3 +111,42 @@ func convertStatsToTF(stats any) (types.String, diag.Diagnostics) {
 	}
 	return types.StringValue(string(statsJSON)), nil
 }
+
+func extractChangeStreamTokenFromStats(stats any) (types.String, diag.Diagnostics) {
+	if stats == nil {
+		return types.StringNull(), nil
+	}
+	var statsMap map[string]interface{}
+
+	statsJSON, err := json.Marshal(stats)
+	if err != nil {
+		return types.StringValue(""), diag.Diagnostics{diag.NewErrorDiagnostic("failed to marshal stats", err.Error())}
+	}
+
+	err = json.Unmarshal(statsJSON, &statsMap)
+	if err != nil {
+		return types.StringValue(""), diag.Diagnostics{diag.NewErrorDiagnostic("failed to unmarshal stats", err.Error())}
+	}
+
+	if data := statsMap["data"]; data != nil {
+		dataMap := data.(map[string]interface{})
+		changeStreamToken, ok := dataMap["changeStreamToken"]
+		if ok {
+			return types.StringValue(changeStreamToken.(string)), nil
+		}
+	}
+
+	return types.StringNull(), nil
+}
+
+func convertPipelineToSdk(pipeline *string) ([]any, diag.Diagnostics) {
+	if pipeline == nil {
+		return nil, diag.Diagnostics{diag.NewErrorDiagnostic("Pipeline is required", "")}
```

Comment:
since `pipeline` is required in the schema, I don't think this is necesssary. I would use `pipeline.ValueString()` directly

Reply:
Done